components/plurals/src/rules/runtime/resolver.rs

```diff
@@ -0,0 +1,61 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::operands::PluralOperands;
+use crate::rules::runtime::ast;
+
+pub fn test_rule(rule: &ast::Rule, operands: &PluralOperands) -> bool {
+    let mut left = true;
+
+    for relation in rule.0.iter() {
+        let relation = relation.as_relation();
+        if left {
+            if relation.and_or == ast::AndOr::Or {
+                return true;
+            } else {
+                left = test_relation(&relation, operands);
+            }
+        } else if relation.and_or == ast::AndOr::Or {
+            left = test_relation(&relation, operands);
+        }
```

Comment:
Optional: I prefer shallower if/else and combining branches where possible.

```suggestion
        if left && relation.and_or == ast::AndOr::Or {
            return true;
        }
        if left || relation.and_or == ast::AndOr::Or {
            left = test_relation(&relation, operands);
        }
```